src/psyclone/psyir/nodes/array_mixin.py

```diff
@@ -598,6 +598,121 @@ def get_outer_range_index(self):
                 return child.position
         raise IndexError
 
+    def same_range(self, index: int, array2, index2: int) -> bool:
```

Comment:
I get:
```
psyir/nodes/array_mixin.py:601:4: R0912: Too many branches (14/12) (too-many-branches)
```
I would be happy to disable this warning with a pylint directive.

If you are touching this file anyway again, could you also add the missing closing ` in lines 136 and 137 (I know it's not your problem, and I am also aware that this function is not documented, since its private, but my check script pointed these lines out to me :) )

Reply:
Done both